pkg/util/url_test.go

```diff
@@ -43,4 +44,36 @@ func TestUrl(t *testing.T) {
 
 		So(result, ShouldEqual, "http://localhost:8080/api/")
 	})
+
+	Convey("When joining two urls where lefthand side has a trailing slash and righthand side has preceding slash", t, func() {
+		result := JoinUrlFragments("http://localhost:8080/", "/api/")
+
+		So(result, ShouldEqual, "http://localhost:8080/api/")
+	})
+}
+
+func TestNewUrlQueryReader(t *testing.T) {
```

Comment:
You're not using goconvey - please fix and/or merge with url test above with multiple    
subsequent Convey statements.